### ISSUE TITLE
Add support for using django's TestClient

### ISSUE THREAD

**Issue by jackton1** [lmcgartland/graphene-file-upload#44]
This should extend [GraphQLTestCase](https://docs.graphene-python.org/projects/django/en/latest/testing/) with support for file mutation test.

> NOTE This is a working code sample
```python
    from django.test import Client, TestCase

    DEFAULT_GRAPHQL_URL = "/graphql/"

    def file_graphql_query(self, query, op_name=None, input_data=None, variables=None, headers=None, files=None, client=None, graphql_url=None):

        if not files:
            raise ValueError('Missing required argument "files": Use `self.query` instead.')
        
        client = client or Client()
        headers = headers or {}
        variables = variables or {}
        graphql_url = graphql_url or DEFAULT_GRAPHQL_URL
        map_ = {}

        for k in files.keys():
            map_[k] = [f'variables.{k}']
            if k not in variables:
                variables[k] = None

        body = {'query': query}
        if op_name:
            body['operationName'] = op_name
        if variables:
            body['variables'] = variables
        if input_data:
            if 'variables' in body:
                body['variables']['input'] = input_data
            else:
                body['variables'] = {'input': input_data}

        data = {
            'operations': json.dumps(body),
            'map': json.dumps(map_),
            **files,
        }
        if headers:
            resp = client.post(graphql_url, data, **headers)
        else:
            resp = client.post(graphql_url, data)

        return resp

```

Usage

```
response = file_query(
            '''
            mutation uploadImageMutation($id: ID!, $image: Upload!) {
                uploadImage(id: $id, image: $image) {
                    user {
                        id
                    }
                }
            }
            ''',
            op_name='uploadImageMutation',
            variables={'id': test_instance.pk},
            files={'image': image_file}
        )
```

TODO:
- [x] Integrate into graphene_file_upload/django package

**Comment by lucas-bremond** [comment]
@jackton1 Pretty cool!

What is the type of `image_file` in this case?

**Comment by jackton1** [comment]
You can use any instance either SimpleUploadedFile https://docs.djangoproject.com/en/3.0/_modules/django/core/files/uploadedfile/ or StringIO object provided it implements a read function

**Comment by lucas-bremond** [comment]
Doing this:

```py
@pytest.mark.django_db
def test_upload (self, client):

    file: SimpleUploadedFile = SimpleUploadedFile(
        name = 'hello.txt',
        content = b'Hello World!',
        content_type = 'text/plain'
    )

    response = file_query(
        client = client,
        query = '''
            mutation uploadFile ($id: UUID!, $file: Upload!) {
                uploadFile (
                    id: $id
                    file: $file
                ) {
                    success
                }
            }
        ''',
        variables = {
            'id': test_instance.id
        },
        files = {
            'file': file
        }
    )

    content: dict = json.loads(response.content)

    assert 'errors' not in content
```

for some reason returns this:

```py
/usr/local/lib/python3.7/site-packages/authservice/tests/utils_for_testing.py:74: in inner
    return f(*args, **kwargs)
/usr/local/lib/python3.7/site-packages/authservice/tests/utils_for_testing.py:74: in inner
    return f(*args, **kwargs)
/usr/local/lib/python3.7/site-packages/django/test/client.py:748: in post
    response = super().post(path, data=data, content_type=content_type, secure=secure, **extra)
/usr/local/lib/python3.7/site-packages/django/test/client.py:405: in post
    secure=secure, **extra)
/usr/local/lib/python3.7/site-packages/django/test/client.py:470: in generic
    return self.request(**r)
/usr/local/lib/python3.7/site-packages/django/test/client.py:716: in request
    self.check_exception(response)
/usr/local/lib/python3.7/site-packages/django/test/client.py:577: in check_exception
    raise exc_value
/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py:47: in inner
    response = get_response(request)
/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py:181: in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
/usr/local/lib/python3.7/site-packages/django/views/decorators/csrf.py:54: in wrapped_view
    return view_func(*args, **kwargs)
/usr/local/lib/python3.7/site-packages/django/views/generic/base.py:70: in view
    return self.dispatch(request, *args, **kwargs)
/usr/local/lib/python3.7/site-packages/django/utils/decorators.py:43: in _wrapper
    return bound_method(*args, **kwargs)
/usr/local/lib/python3.7/site-packages/django/utils/decorators.py:130: in _wrapped_view
    response = view_func(request, *args, **kwargs)
/usr/local/lib/python3.7/site-packages/graphene_django/views.py:155: in dispatch
    data = self.parse_body(request)
/usr/local/lib/python3.7/site-packages/graphene_file_upload/django/__init__.py:20: in parse_body
    request.FILES
/usr/local/lib/python3.7/site-packages/graphene_file_upload/utils.py:22: in place_files_in_operations
    file_obj = files[key]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MultiValueDict: {}>, key = 'file'

    def __getitem__(self, key):
        """
        Return the last data value for this key, or [] if it's an empty list;
        raise KeyError if not found.
        """
        try:
            list_ = super().__getitem__(key)
        except KeyError:
>           raise MultiValueDictKeyError(key)
E           django.utils.datastructures.MultiValueDictKeyError: 'file'

/usr/local/lib/python3.7/site-packages/django/utils/datastructures.py:78: MultiValueDictKeyError
```

I guess I'm using the wrong way...?

**Comment by jackton1** [comment]
Solution for this has been made available in the PR #50
